docs/DeveloperGuide.md

```diff
@@ -247,49 +247,105 @@ tagged with an existing module.
+The following sequence diagram (Figure 2) shows the execution of `countdown days/7`:
 
-2. Add independent event
+![CountdownSequenceDiagram](images/CountdownSequenceDiagram.png)
+Figure 2: Countdown Sequence Diagram
 
-Users can also use Countdown Command to
+When a user executes `CountdownCommand`, `countdown days/7` to count the number of events in the upcoming 7 days from
+from today, `MainWindow` begins the execution of the command. `LogicManager` is then called to execute the
+command. The command is then parsed in `Tr4ckerParser`, and `Tr4ckerParser` will create a new instance of
+`CountdownCommandParser` to parse the user's input. After parsing and checking the validity of user's input, a
+new `CountdownCommand` instance is created. This new instance `countdownCommand` will be passed back to `LogicManager`
+to execute it on `Model` in the `CountdownCommand`. After executing, a new instance `CommandResult`, `commandResult` is
+created. `commandResult` will be passed back to `MainWindow`, then it will be displayed in the results display box in
+`CountdownTabWindow`.
 
-The following activity diagram shows the flow of executing a Countdown Command:
+Within the execution of `countdownCommand`, the list in `Model#getfilteredEventList`, to used to get a list of events
+for `countdownCommand` to work on.
+
+The following activity diagram summarises the flow of executing the various functions of `CountdownCommand`:
 ![CountdownCommandActivityDiagram](images/CountdownActivityDiagram.png)
+Figure 3: Countdown Activity Diagram
+
+The activity diagram shows all the possible paths TR4CKER can take when a user executes a `CountdownCommand`, which is
+identified by the use of the command word, `countdown`. After a command is entered, it's arguments are parsed, to know
+which function the user wants to use:
+1. Add an event
+2. Delete an event
+3. Count the number of events in a certain number of upcoming days
+4. Switch to Countdown tab
 
+The result of the command will be displayed accordingly, either in the events list box, or the result display box at
+near the bottom of the GUI.
 
 #### Design considerations:
 
-##### Aspect 1: Users should be able to easily view the next most recent event
+##### Aspect 1: Viewing events in Countdowns
 
-* **Current Choice:** Use the same `countdown` command to navigate to the next event or previous event. For example,
-`countdown first` would display the earliest upcoming event, `countdown next` will display the event after the one
-currently displayed, and `countdown previous` with display the event before the one currently displayed.
-  * Pros: Users can easily know the chronological sequence of events.
+* **Current Choice:** Display all events in the `upcoming events` list in the Countdown tab, where users can add or
+delete an event and have it be directly reflected in the list.
+  * Pros: Users can see an overview of all events.
+  * Pros: Users mistake in adding any event is easier to spot.
```

Comment:
i think it should be `Users will be able to spot their mistakes in adding any event more easily.`?